examples/blaze/src/main/scala/com/example/http4s/blaze/demo/server/service/GitHubService.scala

```diff
@@ -0,0 +1,55 @@
+package com.example.http4s.blaze.demo.server.service
+
+import cats.effect.Sync
+import cats.syntax.functor._
+import com.example.http4s.blaze.demo.server.endpoints.ApiVersion
+import fs2.Stream
+import io.circe.generic.auto._
+import org.http4s.circe._
+import org.http4s.client.Client
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.{Header, Request, Uri}
+
+// See: https://developer.github.com/apps/building-oauth-apps/authorization-options-for-oauth-apps/#web-application-flow
+class GitHubService[F[_]: Sync](client: Client[F]) extends Http4sClientDsl[F] {
+
+  // NEVER make this data public! This is just a demo!
+  private val ClientId     = "959ea01cd3065cad274a"
+  private val ClientSecret = "53901db46451977e6331432faa2616ba24bc2550"
+
+  private val RedirectUri  = s"http://localhost:8080/$ApiVersion/login/github"
+
+  case class AccessTokenResponse(access_token: String)
+
+  val authorize: Stream[F, Byte] = {
+    val uri = Uri.uri("https://github.com")
+      .withPath("/login/oauth/authorize")
+      .withQueryParam("client_id", ClientId)
+      .withQueryParam("redirect_uri", RedirectUri)
+      .withQueryParam("scopes", "public_repo")
+      .withQueryParam("state", "test_api")
+
+    client.streaming[Byte](Request[F](uri = uri))(_.body)
+  }
+
+  def accessToken(code: String, state: String): F[String] = {
+    val uri = Uri.uri("https://github.com")
```

Comment:
ugh as much as I hate oauth2 I really think it's starting to possibly become a necessity to add a helper.

Reply:
Yeah unfortunately it's required by many companies so I think it'd be nice to have something out of the box like Silhouette provides for Play apps.